internal/fflag/fflag.go

```diff
@@ -0,0 +1,133 @@
+package fflag
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	"github.com/superfly/flyctl/internal/tracing"
+)
+
+const clientSideID string = "6557a71bbffb5f134b84b15c"
+
+type FeatureFlagClient struct {
+	ldContext  ldcontext.Context
+	flags      map[string]FeatureFlag
+	flagsMutex sync.Mutex
+}
+
+type featureFlagClientContextKey struct{}
+
+func NewContextWithClient(ctx context.Context, ffClient *FeatureFlagClient) context.Context {
+	return context.WithValue(ctx, featureFlagClientContextKey{}, ffClient)
+}
+
+func ClientFromContext(ctx context.Context) *FeatureFlagClient {
+	client := ctx.Value(featureFlagClientContextKey{})
+	if client == nil {
+		return nil
+	}
+	return client.(*FeatureFlagClient)
+}
+
+type UserInfo struct {
+	OrganizationID string
+	UserID         int
+}
+
+func NewClient(ctx context.Context, userInfo UserInfo) (*FeatureFlagClient, error) {
+	_, span := tracing.GetTracer().Start(ctx, "new_feature_flag_client")
+	defer span.End()
+
+	orgID, err := strconv.Atoi(userInfo.OrganizationID)
+	if err != nil {
+		orgID = 0
+	}
```

Comment:
I feel like this should probably error if we're passing in something unparseable for an org ID. Maybe default to `0` if a blank string is passed in but otherwise return an error?

Reply:
oh that's way better